File: 152-maximum-product-subarray/maximum-product-subarray.java
class Solution {
    public int maxProduct(int[] nums) {
        int pro = 1, maxi = nums[0];
        for(int i = 0 ; i<nums.length ;i++)
        {
            pro *= nums[i];
            maxi = Math.max(pro,maxi);
            if(pro==0)
            pro = 1 ;
        }
        pro  =1;
        for(int i = nums.length-1 ; i>=0 ;i--)
        {
            pro *= nums[i];
            maxi = Math.max(pro,maxi);
            if(pro==0)
            pro = 1 ;
        }
        return maxi;
    }
}

File: 152-maximum-product-subarray/maximum-product-subarray.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int ans = nums[0], p = 1;
        for(int i = 0; i<nums.size() ;i++)
        {
            p *= nums[i];
            ans = max(ans,p);
            if(p==0)
            p = 1;
        }
        p = 1;
        for(int i = nums.size()-1;i>=0 ;i--)
        {
            p *= nums[i];
            ans = max(ans,p);
            if(p==0)
            p = 1;
        }
        return ans;
    }
};